ebmlite/decoding.py

```diff
@@ -187,11 +187,11 @@ def readString(stream, size):
         @return: The decoded value.
     """
     if size == 0:
-        return b''
+        return u''
 
     value = stream.read(size)
     value = value.partition(b'\x00')[0]
-    return value
+    return str(value, 'ascii')
```

Comment:
I think we want to do one of two things here.  Any value over 127 will raise an exception, probably isn't valid behavior.  We could either use a different codec than ascii which allows the full 0-255 range as valid characters, or we could use non-strict error handling.  Replace would be good since it would keep the character number the same.


https://kb.iu.edu/d/aepu

```suggestion
    return str(value, 'latin-1')
```


https://docs.python.org/3/library/codecs.html#error-handlers

```suggestion
    return str(value, 'ascii', 'replace')
```

Reply:
I don't think it should raise an exception... This is decoding of data which may not have originated with the user doing the reading, so they aren't necessarily in a position to fix it. Replacing non-ASCII characters and generating a warning would be better.

Reply:
As discussed with @CFlaniganMide separately, [the spec](https://datatracker.ietf.org/doc/html/rfc8794#section-7.4) dictates:

> A String Element MUST either be empty (zero-length) or contain
   printable ASCII characters ... in the range of 0x20 to 0x7E,
   with an exception made for [null-]termination...

I think it's fair to have it raise a warning, and not error and fail on bad data. If a user *does* want a hard failure, they can change warnings into errors via `with warnings.catch_warnings(): warnings.simplefilter("error", category=UnicodeWarning)`. If they don't, they can do something similar to ignore the warnings.

In 208d44dedc2eac05192836bb02ea62b3ae13df45 I implemented the warn-instead-of-fail suggestion; let me know if this works for you.